core/src/main/scala/cats/MonadReader.scala

```diff
@@ -0,0 +1,14 @@
+package cats
+
+/** A monad that has the ability to read from an environment. */
+trait MonadReader[F[_, _], R] extends Monad[F[R, ?]] {
+  /** Get the environment */
+  def ask: F[R, R]
+
+  /** Modify the environment */
+  def local[A](f: R => R)(fa: F[R, A]): F[R, A]
```

Comment:
I am OK with either, one thing I just noticed is that the direction may or may not look nicer due to the "flow" of the operation. For instance,

```scala
[M[_], A]def flatMap[B](ma: M[A])(f: A => M[B]): M[B]
```
We run the effects of `ma` first, and then apply `f` to it, and then run the effects of `M[B]`, so the flow is left to right. similarly for `map` and `traverse`.

With local:

```scala
[F[_, _], R]def local[A](f: R => R)(fa: F[R, A]): F[R, A]
```

We are first modifying the environment with `f` and then running `fa` against that, as opposed to running `fa` and then modifying the environment after the fact.